src/hooks/use-certificate-trust.ts

```diff
@@ -1,35 +1,27 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useWindowListener } from 'src/hooks/use-window-listener';
-import { getIpcApi } from 'src/lib/get-ipc-api';
+import { checkCertificateTrust, selectIsRootCATrusted } from 'src/store/slices/certificate-trust';
+import { useAppDispatch, useRootSelector } from 'src/stores';
 
 /**
  * Custom hook that checks if the Studio CA certificate is trusted on the system
  * @returns A boolean indicating if the certificate is trusted
  */
 export function useCertificateTrust(): boolean {
-	const [ isTrusted, setIsTrusted ] = useState< boolean >( false );
+	const dispatch = useAppDispatch();
```

Comment:
@fredrikekelund, I have been thinking about your comment, and I was also unhappy with how `useCertificateTrust()` was implemented. It was creating a flicker issue when changing sites, so I went ahead and implemented it using the Redux store.

This is the first time I have implemented it. Let me know what you think about it.

Reply:
Just to clarify, my comment https://github.com/Automattic/studio/pull/1117#discussion_r2073151154 was not meant to say that the current approach was wrong per se (though I generally support refactoring this using Redux Toolkit). I mostly wanted to point out that using an `isMounted` ref might not be such a bad idea.

Reply:
I agree. I also don't mean to say that the previous approach was bad; I just wasn't happy with the flicker issue.